R/add_difference_row.R

```diff
@@ -0,0 +1,435 @@
+#'     method = "glm",
+#'     method.args = list(family = binomial)
+#'   ) |>
+#'     cardx::ard_regression_basic(exponentiate = TRUE) |>
+#'     dplyr::filter(.data$variable == .env$by)
+#' }
+#'
+#' trial |>
+#'   tbl_summary(by = trt, include = response, missing = "no") |>
+#'   add_stat_label() |>
+#'   add_difference_row(
+#'     reference = "Drug A",
+#'     statistic = everything() ~ c("{estimate}", "{conf.low}, {conf.high}", "{p.value}"),
+#'     test = everything() ~ ard_odds_ratio,
+#'     estimate_fun = everything() ~ label_style_ratio()
+#'   ) |>
+#'   # change the default label for the 'Odds Ratio'
+#'   modify_table_body(
+#'     ~ .x |>
+#'       dplyr::mutate(
+#'         label = ifelse(label == "Coefficient", "Odds Ratio", label)
+#'       )
+#'   ) |>
+#'   # add footnote about logistic regression
+#'   modify_footnote_body(
+#'     footnote = "Age-adjusted logistic regression model",
+#'     column = "label",
+#'     rows = variable == "response-row_difference"
+#'   )
+add_difference_row.tbl_summary <- function(x,
+                                           reference,
+                                           statistic = everything() ~ "{estimate}",
+                                           test = NULL,
+                                           group = NULL,
+                                           header = NULL,
+                                           adj.vars = NULL,
+                                           test.args = NULL,
+                                           conf.level = 0.95,
+                                           include = everything(),
+                                           pvalue_fun = label_style_pvalue(digits = 1),
+                                           estimate_fun = list(
+                                             c(all_continuous(), all_categorical(FALSE)) ~ label_style_sigfig(),
+                                             all_dichotomous() ~ label_style_sigfig(scale = 100, suffix = "%"),
+                                             all_tests("smd") ~ label_style_sigfig()
+                                           ),
+                                           ...) {
+  # check inputs ---------------------------------------------------------------
+  set_cli_abort_call()
+  check_dots_empty(call = get_cli_abort_call())
+  updated_call_list <- c(x$call_list, list(add_difference = match.call()))
+  check_not_missing(reference)
+  check_scalar(reference)
+
+  # checking that input x has a by var and it has two levels
+  if (is_empty(x$inputs$by)) {
+    "Cannot run {.fun add_difference_row} when {.code tbl_summary()} does not include a {.arg by} argument." |>
+      cli::cli_abort(call = get_cli_abort_call())
+  }
+
+  # if `pvalue_fun` not modified, check if we need to use a theme p-value
+  if (missing(pvalue_fun)) {
+    pvalue_fun <-
+      get_theme_element("pkgwide-fn:pvalue_fun") %||%
+      pvalue_fun
```

Comment:
Can it remain missing if there is nothing in the theme?

Reply:
I guess you need to actively take it out, but having this inside if(missing()) makes the %||% risky maybe?

Reply:
The `missing()` function is named poorly. It's really testing if the default value has been changed by the user. What we are saying here is that if the user has _not_ specified the `pvalue_fun` argument, then we use the theme pvalue function, and when that is not set, we land back on the default value for the argument.

Reply:
Understood! thanks